resolver/resolver.cc

```diff
@@ -3643,6 +3695,40 @@ class ResolveSignaturesWalk {
             [&](const ast::ExpressionPtr &e) {});
     }
 
+    static bool hasCompatibleOverloadedSigsWithKwArgs(core::Context ctx, int numSigs, const SigInformation &info,
+                                                      const std::vector<ArgInformation> &args) {
+        if (numSigs != 2) {
+            return false;
+        }
+        if (!info.allArgsMatched) {
+            return false;
+        }
+        if (info.hasMissingArgument) {
+            return false;
+        }
+
+        const auto &argv0 = args[0];
+        const auto &argv1 = args[1];
+
+        // Unlike std::equal, absl::c_equal will test for equal sizes.
+        auto argsEqual = [&ctx](const auto &arg0, const auto &arg1) {
+            if (arg0.name != arg1.name) {
+                return false;
+            }
+            return core::Types::equiv(ctx, arg0.type, arg1.type);
```

Comment:
🤔 Do you mind adding some tests for overloads that have generics (both `T.type_parameter` and `type_member`) in them?

```ruby
class A
  extend T::Sig

  sig do
    type_parameters(:U)
      .params(x: T.type_parameter(:U))
      .void
  end
  sig do
    type_parameters(:V)
      .params(x: T.type_parameter(:V), blk: T.proc.void)
      .void
  end
  def example(x, &blk); end
end
```

Reply:
^ as a minimum, but maybe you can think of other test cases that use generics in combination with `T.any` / `T.all` etc.